p2p/ipld/read.go

```diff
@@ -0,0 +1,96 @@
+package ipld
+
+import (
+	"context"
+	"errors"
+	"math"
+
+	"github.com/ipfs/go-cid"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+// /////////////////////////////////////
+//	Get Leaf Data
+// /////////////////////////////////////
+
+// GetLeafData fetches and returns the data for leaf leafIndex of root rootCid.
+// It stops and returns an error if the provided context is cancelled before
+// finishing
+func GetLeafData(
+	ctx context.Context,
+	rootCid cid.Cid,
+	leafIndex uint32,
+	totalLeafs uint32, // this corresponds to the extended square width
+	api coreiface.CoreAPI,
+) ([]byte, error) {
+	// calculate the path to the leaf
+	leafPath, err := calcCIDPath(leafIndex, totalLeafs)
+	if err != nil {
+		return nil, err
+	}
+
+	// use the root cid and the leafPath to create an ipld path
+	p := path.Join(path.IpldPath(rootCid), leafPath...)
+
+	// resolve the path
+	node, err := api.ResolveNode(ctx, p)
+	if err != nil {
+		return nil, err
+	}
+
+	// return the leaf, without the nmt-leaf-or-node byte
+	return node.RawData()[1:], nil
+}
+
+func calcCIDPath(index, total uint32) ([]string, error) {
```

Comment:
nit: CIDPath might a bit unspecific (?). Should we call this: `leafIPLDPath` or simply `leafPath`?

Reply:
as always, thanks for the feedback! resolved https://github.com/lazyledger/lazyledger-core/pull/212/commits/f9c49539a5652ca830242ec325d07540405e6719